### ISSUE TITLE
Improving SARIF format for Github integration

### ISSUE THREAD

**Issue by h3xstream** [spotbugs/spotbugs#1630]
The SARIF format was a nice addition. I think many integrations will use it in the future.
The most common one currently will probably be Github.

Base on few tests, it appears that Github requires The field "text" to be present in **results > message > text**

From the REST webservice, it is giving an obscure error `"error": "expected a result message,expected a result message,expected a result message,[...]"`

From the code scanning section, the error "Analysis processing failed" is given.
![test](https://user-images.githubusercontent.com/593130/127538972-18d8c5d6-9b85-49ec-b65f-562a89b9c302.png)

In summary, Github does not take into account the `ruleId` field. The description need to be in every result apparently.  
It requires `message.text`:
Ref: https://docs.github.com/en/code-security/secure-coding/integrating-with-code-scanning/sarif-support-for-code-scanning#result-object

**Comment by h3xstream** [comment]
I will open a PR with the proposition to fix the missing field.

**Comment by KengoTODA** [comment]
closed by #1631
